src/requests/index.ts

```diff
@@ -21,18 +27,38 @@ import type { GraphQLClient } from "graphql-request";
 const defaultURL = process.env.HASURA_GRAPHQL_URL;
 
 export class CoreDomainClient {
-  client: GraphQLClient;
+  client!: GraphQLClient;
```

Comment:
the `!` lets TS know that I'm handling setting this attribute outside of the constructor (in this case, in `setClient`.